pkg/kubectl/resource/builder.go

```diff
@@ -757,3 +774,17 @@ func HasNames(args []string) (bool, error) {
 	}
 	return hasCombinedTypes || len(args) > 1, nil
 }
+
+func (b *Builder) validateNamespace(c RESTClient, namespace string) error {
+	if !b.validateNamespaces {
+		return nil
+	}
+	if namespace == api.NamespaceDefault || namespace == api.NamespaceAll {
+		return nil
+	}
+	err := c.Get().AbsPath("api", "v1", "namespaces", namespace).Do().Error()
```

Comment:
@smarterclayton I don't think we want to do this server side because it implies an extra lookup for every API request, which is a cost that I don't think we want to pay just for better UX.